config/v1/types_dns.go

```diff
@@ -53,6 +53,13 @@ type DNSSpec struct {
 	//
 	// +optional
 	PrivateZone *DNSZone `json:"privateZone,omitempty"`
+	// platform holds configuration specific to the underlying
+	// infrastructure provider for DNS.
+	// When omitted, this means the user has no opinion and the platform is left
+	// to choose reasonable defaults. These defaults are subject to change over time.
+	//
+	// +optional
+	Platform *DNSPlatformSpec `json:"platform,omitempty"`
```

Comment:
For our configuration APIs, we avoid pointers for discoverability and make the zero value act as "no opinion, do good things"